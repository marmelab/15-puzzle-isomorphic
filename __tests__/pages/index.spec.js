import React from 'react';
import Index from '../../pages/index';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

describe('Index', () => {
    it('it should render Index without error', () => {
        renderer.create(<Index />);
    });

    it('should contain the main title', () => {
        const app = shallow(<Index />);
        expect(app.find('h1').text()).toEqual('Welcome to the 15 puzzle game!');
    });

    it('should render the Index correctly', () => {
        const component = renderer.create(<Index />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
