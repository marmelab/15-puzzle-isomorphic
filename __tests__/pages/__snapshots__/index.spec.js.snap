// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match with snapshot 1`] = `
<div>
  <h1>
    Welcome to the 15 puzzle game!
  </h1>
</div>
`;

exports[`Index should render the Index correctly 1`] = `
<div>
  <h1>
    Welcome to the 15 puzzle game!
  </h1>
</div>
`;
